### PR TITLE
Replace deprecated code and fix issues found by static code analysis

### PR DESCRIPTION
**Pull request by susnux**
* `OCP\IInitialStateService` is deprecated in favor of `OCP\AppFramework\Services\IInitialState`
* Remove unused `IGroupManager` in `PageController`
* `QueryBuilder::execute` is deprecated
* Fix psalm detected issues / static code analysis
* Minor other issues

Also added a psalm CI workflow for static code analysis.

### CODE REVIEW

File: psalm.xml
Comment (jotoeri) [inline psalm.xml:1]: Where's that file from? Can you check the diff to the recommended file in the docs?
https://docs.nextcloud.com/server/stable/developer_manual/digging_deeper/continuous_integration.html#static-analysis

File: lib/Service/SubmissionService.php [hunk 1]
```diff
@@ -174,13 +171,15 @@ public function writeCsvToCloud(string $hash, string $path): string {
...
+			/** @var \OCP\Files\Folder $node */
 			$file = $node->get($csvData['fileName']);
 		} catch (\OCP\Files\NotFoundException $e) {
 			$node->newFile($csvData['fileName']);
 			$file = $node->get($csvData['fileName']);
 		}
 
 		// Write the data to file
+		/** @var \OCP\Files\File $file */
 		$file->putContent($csvData['data']);
```
Comment (jotoeri) [inline lib/Service/SubmissionService.php:183]: Place these comments at where the variables they are defined, not where they are used?

Comment (susnux) [reply lib/Service/SubmissionService.php]: Done

**Review by jotoeri — APPROVED** [review]

**Review by Chartman123 — APPROVED** [review]

**Pull request merged**